lib/puppetserver/utils/refactored_client.rb

```diff
@@ -0,0 +1,108 @@
+require 'openssl'
+require 'net/https'
+
+module Puppetserver
+  module Utils
+    # Utilities for doing HTTPS against the CA that wraps Net::HTTP constructs
+    class RefactoredClient
```

Comment:
Are we going to wait to rename this until the refactor ticket?

Reply:
@Magisus I'm doing the refactor ticket now.  So either way I'll make sure the work gets done.